mongorestore/mongorestore_test.go

```diff
@@ -583,3 +583,91 @@ func TestFixHashedIndexes(t *testing.T) {
 		})
 	})
 }
+
+func TestRandom(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	ctx := context.TODO()
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	Convey("Test MongoRestore with {autoIndexId: false} in a non-local database's collection", t, func() {
+		dbName := session.Database("testdata")
+
+		defer dbName.Collection("test_auto_idx").Drop(ctx) // drop the collection to clean up resources
+
+		var args []string
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		restore.TargetDirectory = "testdata/test_auto_idx.bson"
+		result := restore.Restore()
+		So(result.Err, ShouldBeNil)
+
+		// Find the collection
+		filter := bson.D{{"name", "test_auto_idx"}}
+		cursor, err := session.Database("testdata").ListCollections(ctx, filter)
+		if err != nil {
+			panic(err)
+		}
+		defer cursor.Close(ctx)
+		if !cursor.Next(ctx) {
+			panic("no matching collection found")
+		}
+		var collInfo struct {
+			Options bson.M
+		}
+		if err = cursor.Decode(&collInfo); err != nil {
+			panic(err)
+		}
+
+		Convey("{autoIndexId: false} should be flipped to true iff server version >= 4.0", func() {
+			if restore.serverVersion.GTE(db.Version{4, 0, 0}) {
+				So(collInfo.Options["autoIndexId"], ShouldBeTrue)
+			} else {
+				So(collInfo.Options["autoIndexId"], ShouldBeFalse)
+			}
+		})
+	})
+
+	Convey("Test MongoRestore with {autoIndexId: false} in a local database's collection", t, func() {
+		dbName := session.Database("local")
+
+		defer dbName.Collection("test_auto_idx").Drop(ctx) // drop the collection to clean up resources
+
+		var args []string
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		restore.TargetDirectory = "local/test_auto_idx.bson"
+		result := restore.Restore()
+		So(result.Err, ShouldBeNil)
+
+		// Find the collection
+		filter := bson.D{{"name", "test_auto_idx"}}
+		cursor, err := session.Database("local").ListCollections(ctx, filter)
+		if err != nil {
+			panic(err)
+		}
+		defer cursor.Close(ctx)
+		if !cursor.Next(ctx) {
+			panic("no matching collection found")
+		}
+		var collInfo struct {
+			Options bson.M
+		}
+		if err = cursor.Decode(&collInfo); err != nil {
+			panic(err)
+		}
+
+		Convey("{autoIndexId: false} should never be flipped to true", func() {
+			if restore.serverVersion.LTE(db.Version{4, 2, 0}) {
```

Comment:
do we need the version check here?

Reply:
I figured this would be a sanity check to make sure that the behavior we added for 4.0 and 4.2 is not applied in the case that the database is `local`.

Reply:
but even so, this checks all the version <=4.2 instead of 4.0 and 4.2?